drivers/clock_control/clock_control_it8xxx2.c

```diff
@@ -0,0 +1,431 @@
+/*
+ * Copyright (c) 2022 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT ite_it8xxx2_ecpm
+
+#include <soc.h>
+#include <zephyr/logging/log.h>
+#include "soc_espi.h"
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/dt-bindings/clock/it8xxx2_clock.h>
+#include <zephyr/drivers/clock_control/it8xxx2_clock_control.h>
+
+LOG_MODULE_REGISTER(clock_control_it8xxx2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define PLLFREQ_MASK BIT_MASK(4)
+#define MCUFREQ_MASK BIT_MASK(3)
+#define EC_FREQ_MASK BIT_MASK(4)
+#define SMBFREQ_MASK BIT_MASK(4)
+#define FNDFREQ_MASK GENMASK(7, 4)
+
+struct clock_control_it8xxx2_config {
+	struct ecpm_it8xxx2_regs *reg_base;
+	int pll_freq;
+};
+
+const struct clock_control_it8xxx2_config *clock_config;
+
+/* ITE IT8XXX2 ECPM Functions */
+static inline int it8xxx2_clock_control_on(const struct device *dev,
+					clock_control_subsys_t sub_system)
+{
+	const struct clock_control_it8xxx2_config *cfg = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = cfg->reg_base;
+	struct it8xxx2_clock_control_cells *clk_cfg =
+		(struct it8xxx2_clock_control_cells *)(sub_system);
+
+	uint8_t offset = clk_cfg->offset;
+	uint8_t code = clk_cfg->bitmask;
+	uint8_t always = clk_cfg->always;
+
+	volatile uint8_t *reg =
+		(volatile uint8_t *)((uint32_t)ecpm_regs + (uint32_t)offset);
+
+	*reg = (*reg & ~code) | always;
+
+	return 0;
+}
+
+static inline int it8xxx2_clock_control_off(const struct device *dev,
+					clock_control_subsys_t sub_system)
+{
+	const struct clock_control_it8xxx2_config *cfg = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = cfg->reg_base;
+	struct it8xxx2_clock_control_cells *clk_cfg =
+		(struct it8xxx2_clock_control_cells *)(sub_system);
+
+	uint8_t offset = clk_cfg->offset;
+	uint8_t code = clk_cfg->bitmask;
+	uint8_t always = clk_cfg->always;
+
+	volatile uint8_t *reg =
+		(volatile uint8_t *)((uint32_t)ecpm_regs + (uint32_t)offset);
+
+	*reg |= (always | code);
+
+	return 0;
+}
+
+const uint32_t pll_reg_to_freq[8] = {MHZ(8), MHZ(16), MHZ(24), MHZ(32),
+				MHZ(48), MHZ(64), MHZ(72), MHZ(96)};
```

Comment:
please format code properly, you can try with clang-format.

Reply:
Thank you, sir. I'll try with the clang-format.

Reply:
Done with clang-format. Thanks again for the information and guidance.